[GCP FileShare Support] Invalid memory address or nil pointer dereference for GCP Fileshare Updation

**Issue/Feature Description:** GCP Fileshare Updation is failing with the following reason:

Invalid memory address or nil pointer dereference
![image](https://user-images.githubusercontent.com/8712951/94366797-20f8d580-00f8-11eb-8f37-963c6158218d.png)
Error logs:

```
[2020-09-27T19:25:51+05:30] [ERROR] [/root/gopath/src/github.com/opensds/multi-cloud/file/pkg/service/service.go] [service.(*fileService).UpdateFileShare():295] [PID:1] Received error in creating file shares at backend patch operation failed: googleapi: Error 429: Quota limit 'StandardStorageGbPerRegion' has been exceeded. Limit: 2048 in region asia-east1., rateLimitExceeded
2020-09-27 19:25:51  file=grpc.go:372 level=error panic recovered: runtime error: invalid memory address or nil pointer dereference
2020-09-27 19:25:51  file=grpc.go:373 level=error goroutine 813 [running]:
runtime/debug.Stack(0xc0000be480, 0xc000bc4e02, 0xc000bc4e20)
        /usr/local/go/src/runtime/debug/stack.go:24 +0x9d
github.com/micro/go-micro/v2/server/grpc.(*grpcServer).processRequest.func1.1(0xc000a07250)
        /root/gopath/pkg/mod/github.com/micro/go-micro/v2@v2.6.0/server/grpc/grpc.go:373 +0x2cf
panic(0x1838560, 0x2c432c0)
        /usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/opensds/multi-cloud/file/pkg/service.(*fileService).UpdateFileShare(0xc0005f2b40, 0x1ef0b00, 0xc0007301e0, 0xc000276d40, 0xc000730090, 0x4d00ec, 0x1a02880)
        /root/gopath/src/github.com/opensds/multi-cloud/file/pkg/service/service.go:296 +0x566
github.com/opensds/multi-cloud/file/proto.(*fileHandler).UpdateFileShare(0xc000219660, 0x1ef0b00, 0xc0007301e0, 0xc000276d40, 0xc000730090, 0x7fc0abb95d98, 0x0)
        /root/gopath/src/github.com/opensds/multi-cloud/file/proto/file.pb.micro.go:157 +0x5b
reflect.Value.call(0xc000278a80, 0xc00000e930, 0x13, 0x1b16bec, 0x4, 0xc0009871b0, 0x4, 0x4, 0x16, 0x1a26680, ...)
        /usr/local/go/src/reflect/value.go:460 +0x5f6
reflect.Value.Call(0xc000278a80, 0xc00000e930, 0x13, 0xc0009871b0, 0x4, 0x4, 0x203000, 0xc0009871b0, 0x40c1d6)
        /usr/local/go/src/reflect/value.go:321 +0xb4
github.com/micro/go-micro/v2/server/grpc.(*grpcServer).processRequest.func1(0x1ef0b00, 0xc0007301e0, 0x1f06f80, 0xc0009bea00, 0x1a02880, 0xc000730090, 0x0, 0x0)
        /root/gopath/pkg/mod/github.com/micro/go-micro/v2@v2.6.0/server/grpc/grpc.go:378 +0x303
github.com/micro/go-micro/v2/util/wrapper.AuthHandler.func1.1(0x1ef0b00, 0xc000730180, 0x1f06f80, 0xc0009bea00, 0x1a02880, 0xc000730090, 0xc000730180, 0xc000107030)
        /root/gopath/pkg/mod/github.com/micro/go-micro/v2@v2.6.0/util/wrapper/wrapper.go:282 +0x5e3
github.com/micro/go-micro/v2/util/wrapper.TraceHandler.func1.1(0x1ef0a40, 0xc000276d00, 0x1f06f80, 0xc0009bea00, 0x1a02880, 0xc000730090, 0x16, 0x1a02880)
        /root/gopath/pkg/mod/github.com/micro/go-micro/v2@v2.6.0/util/wrapper/wrapper.go:122 +0x1c6
github.com/micro/go-micro/v2/util/wrapper.HandlerStats.func1.1(0x1ef0a40, 0xc000276d00, 0x1f06f80, 0xc0009bea00, 0x1a02880, 0xc000730090, 0x14, 0x0)
        /root/gopath/pkg/mod/github.com/micro/go-micro/v2@v2.6.0/util/wrapper/wrapper.go:67 +0x83
github.com/micro/go-micro/v2/server/grpc.(*grpcServer).processRequest(0xc00047c900, 0x1efc8a0, 0xc000886240, 0xc00065a9c0, 0xc0004b8360, 0xc000b86700, 0x16, 0x1ef0a40, 0xc000276d00, 0x1ef0ac0, ...)
        /root/gopath/pkg/mod/github.com/micro/go-micro/v2@v2.6.0/server/grpc/grpc.go:395 +0x7ee
github.com/micro/go-micro/v2/server/grpc.(*grpcServer).handler(0xc00047c900, 0x0, 0x0, 0x1efc8a0, 0xc000886240, 0x0, 0x0)
        /root/gopath/pkg/mod/github.com/micro/go-micro/v2@v2.6.0/server/grpc/grpc.go:314 +0x77f
google.golang.org/grpc.(*Server).processStreamingRPC(0xc000239680, 0x1f06fe0, 0xc000838c00, 0xc00067ea00, 0x0, 0xc0005f2a60, 0x0, 0x0, 0x0)
        /root/gopath/pkg/mod/google.golang.org/grpc@v1.26.0/server.go:1237 +0xb42
google.golang.org/grpc.(*Server).handleStream(0xc000239680, 0x1f06fe0, 0xc000838c00, 0xc00067ea00, 0x0)
        /root/gopath/pkg/mod/google.golang.org/grpc@v1.26.0/server.go:1323 +0xc0a
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc000bba4f0, 0xc000239680, 0x1f06fe0, 0xc000838c00, 0xc00067ea00)
        /root/gopath/pkg/mod/google.golang.org/grpc@v1.26.0/server.go:722 +0xbb
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /root/gopath/pkg/mod/google.golang.org/grpc@v1.26.0/server.go:720 +0xa1
```

**Why this issue to fixed / feature is needed(give scenarios or use cases):**  To Support GCP FileShare Updation with proper Error Handling

**How to reproduce, in case of a bug:**

**Other Notes /  Environment Information: (Please give the env information, log link or any useful information for this issue)**

--- a/file/pkg/service/service.go
+++ b/file/pkg/service/service.go
@@ -292,7 +292,10 @@ func (f *fileService) UpdateFileShare(ctx context.Context, in *pb.UpdateFileShar
 
 	fs, err := sd.UpdatefileShare(ctx, in)
 	if err != nil {
-		log.Errorf("Received error in creating file shares at backend %s", err)
+		log.Errorf("Received error in updating file shares at backend %s", err)
+		if fs == nil {
+			return err
+		}
 		fs.Fileshare.Status = utils.FileShareStateError
 	} else {
 		fs.Fileshare.Status = utils.FileShareStateUpdating

--- a/file/pkg/service/service.go
+++ b/file/pkg/service/service.go
@@ -292,7 +292,7 @@ func (f *fileService) UpdateFileShare(ctx context.Context, in *pb.UpdateFileShar
 
 	fs, err := sd.UpdatefileShare(ctx, in)
 	if err != nil {
-		log.Errorf("Received error in updating file shares at backend %s", err)
+		log.Errorf("received error in updating file shares at backend %s", err)
 		if fs == nil {
 			return err
 		}

